android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt

```diff
@@ -101,6 +100,22 @@ constructor(
     configurationRegistry.retrieveConfiguration(ConfigType.Navigation)
   }
 
+  fun retrieveRemoteIconsBitmap() {
+    navigationConfiguration.clientRegisters
+      .filterNot { it.menuIconConfig == null }
+      .filter { it.menuIconConfig!!.type == "remote" }
+      .forEach {
+        val resourceId = it.menuIconConfig?.reference!!.substringAfter("/")
+        viewModelScope.launch {
+          val byteArray = registerRepository.loadResource<Binary>(resourceId)?.data
+          val decodedByteArray = Base64.decode(byteArray, Base64.DEFAULT)
+          Timber.d(decodedByteArray.contentToString())
+          val bitmap = decodedByteArray?.decodeToBitmap()
+          it.menuIconConfig?.decodedBitmap = bitmap
```

Comment:
I think we can inline this.
```suggestion
          it.menuIconConfig?.decodedBitmap = decodedByteArray?.decodeToBitmap()
```

Reply:
Agree